File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/School.java
package com.raxim.myscoutee.profile.data.document.mongo;

import java.util.Date;
import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Document(collection = "schools")
public class School implements Cloneable {
    @Id
    @JsonProperty(value = "key")
    private UUID id;

    @JsonProperty(value = "type")
    private String type;

    @JsonProperty(value = "name")
    private String name;

    @JsonProperty(value = "locationKey")
    private String locationKey;

    @JsonProperty(value = "range")
    private RangeLocal range;

    @JsonProperty(value = "role")
    private String role;

    @JsonIgnore
    private Date createdDate;

    @JsonIgnore
    private String status;

    public School(RangeLocal range) {
        this.range = range;
    }

    public School() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLocationKey() {
        return locationKey;
    }

    public void setLocationKey(String locationKey) {
        this.locationKey = locationKey;
    }

    public RangeLocal getRange() {
        return range;
    }

    public void setRange(RangeLocal range) {
        this.range = range;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public Object clone()
            throws CloneNotSupportedException {
        return super.clone();
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/School.kt
package com.raxim.myscoutee.profile.data.document.mongo

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.querydsl.core.annotations.QueryEntity
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

@QueryEntity
@Document(collection = "schools")
data class School(
    @Id
    @field:JsonProperty(value = "key")
    val id: UUID? = UUID.randomUUID(),

    @field:JsonProperty(value = "type")
    val type: String? = null,
    @field:JsonProperty(value = "name")
    val name: String? = null,
    // google
    @field:JsonProperty(value = "locationKey")
    val locationKey: String? = null,

    @field:JsonProperty(value = "range")
    val range: RangeLocal? = RangeLocal(),
    @field:JsonProperty(value = "role")
    val role: String? = null,

    @field:JsonIgnore
    var createdDate: Date? = Date(),

    // Active (A), Deleted (D)
    @field:JsonIgnore
    var status: String? = "A"
)